libvast/src/table_index.cpp

```diff
@@ -90,27 +104,34 @@ caf::error table_index::add(const table_slice_ptr& x) {
   VAST_ASSERT(x != nullptr);
   VAST_ASSERT(x->layout() == layout());
   VAST_TRACE(VAST_ARG(x));
+  // Store IDs of the new rows.
+  auto first = x->offset();
+  auto last = x->offset() + x->rows();
+  VAST_ASSERT(first < last);
+  if (first >= row_ids_.size()) {
+    row_ids_.append_bits(false, first - row_ids_.size());
+    row_ids_.append_bits(true, last - first);
+  } else {
+    ids tmp;
+    tmp.append_bits(false, first);
+    tmp.append_bits(true, last - first);
+    row_ids_ |= tmp;
```

Comment:
I would prefer not to use names like `tmp`, because they don't carry any semantics. Something like `new_ids` in this case would be more meaningful.

Reply:
Upleveling, I have trouble understanding when this branch is taken. Say I have two tables slices `x` and `y`, where `y` arrives after `x`. Then the append-only nature of bitmap indexes requires that `y` has greater IDs than `x`.

Reply:
I turned the `if` into an assertion and removed the `else` branch.